pkg/ipmi/dev.go

```diff
@@ -13,6 +13,69 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+func (cc CompletionCode) Error() string {
```

Comment:
It's a little non-idiomatic to make a CompletionCode an error. For one, an error should always be declared as `...Error`.

Here's what I suggest:

```go
func (cc CompletionCode) String() string

type CompletionError CompletionCode

func (ce CompletionError) Error() string {
  return fmt.Sprintf("command completed with non-OK code: %s", CompletionCode(ce))
}
```

and then below, you can

```
rerr = fmt.Errorf("invalid response: %w", CompletionError(cc))
```

when you get the response, for testing you can do e.g. `errors.Is(err, ipmi.CompletionError(ipmi.IPMI_CC_TIMEOUT))`

Reply:
Done.

Reply:
Missing the rerr = fmt.Errorf part

Reply:
Ah, I deliberately left that out. We already add context to the error in `CompletionError`'s `Error()` method. I don't think we need to wrap it again when returning it.